actions/admin/upgrade.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * Run a upgrade on a batch of items
+ *
+ * Run for 2 seconds per request as set by $batch_run_time_in_secs.
+ * This includes the engine loading time.
+ */
+
+// from engine/start.php
+global $START_MICROTIME;
+
+$offset = (int) get_input('offset', 0);
+
+$batch_run_time_in_secs = 2;
+
+$class_name = get_input('class_name');
+
+$upgrade = new $class_name;
```

Comment:
Check if class exists first? Validate that the class name hasn't been altered client side (hmac perhaps)?